### ISSUE TITLE
Overlaying rotated objects causes artifacts (kivy 2.2.0 only)

### ISSUE THREAD

**Issue by 20Tauri** [kivy/kivy#8254]
**Software Versions**
* Python: 3.8.10
* OS: Window 10
* Kivy: 2.2.0
* Kivy installation method: pip

**Describe the bug**
When a widget is rotated and another widget overlaps it, the displayed result is incorrect.

The rotation method is classical and a result is displayed later in the description

**Expected behavior**
Same with 2.1.0: no artefact

**Code and Logs and screenshots**
```python
    canvas.before:
        PushMatrix
        Rotate:
            angle: self.rotate_angle
            axis: 0, 0, 1
            origin: self.center

    canvas.after:
        PopMatrix
```

Problem is on the left top corner and the bottom right corner
Note that the same object in the left bottom corner is displayed correctly.

![image](https://github.com/kivy/kivy/assets/12613719/0da8a19d-d5bf-4cc2-ae22-29c406aacef2)

**Comment by 20Tauri** [comment]
Caused https://github.com/kivy/kivy/pull/8169 and Stencil in style.kv (see https://github.com/kivy/kivy/pull/8169/commits/b8910a718cef61ab12a9774fd2e65c9bf76c21fc)

**Comment by DexerBR** [comment]
Hi @20Tauri!

Could you test the master (which includes PR #8276)?

**Comment by 20Tauri** [comment]
Hi @DexerBR, 

Tests have been made with a 2.2.0 version with manual report of your change
Seems OK,

I have opened a new issue (https://github.com/kivy/kivy/issues/8278) because I have to override style.kv to mae my application work properly... This not relate our problem

**Comment by misl6** [comment]
Fixed via #8276
